src/ansys/hps/client/client.py

```diff
@@ -195,12 +222,25 @@ def __init__(
             self.refresh_token = tokens.get("refresh_token", None)
 
         parsed_username = None
-
+        token = {}
         try:
-            parsed_jwt = jwt.decode(self.access_token, options={"verify_signature": False})
-            parsed_username = parsed_jwt["preferred_username"]
-        except:
-            log.warning("Could not retrieve preferred_username from access token.")
+            token = jwt.decode(self.access_token, options={"verify_signature": False})
+        except Exception:
+            raise HPSError("Authentication token was invalid.")
+
+        # Try to get the standard keycloak name, then other possible valid names
+        parsed_username = token.get("preferred_username", None)
+        if not parsed_username:
+            parsed_username = token.get("username", None)
+        if not parsed_username:
+            parsed_username = token.get("name", None)
+
+        # Service accounts look like "aud -> service_client_id"
+        if not parsed_username:
+            if token.get("oid", "oid_not_found") == token.get("sub", "sub_not_found"):
+                parsed_username = "service_account_" + token.get("aud", "aud_not_set")
+            else:
+                raise HPSError("Authentication token had no username.")
```

Comment:
Let's also move all this username parsing into a separate function. By the way, do we really need to do this matching?

Reply:
I'm not actually sure, I'm finding the name just because we already did it.  I think I might use it in scaler somewhere but if no-one needs it, then I don't have an issue to remove the entire scheme?  What are your thoughts?